src/erlcloud_workspaces.erl

```diff
@@ -0,0 +1,446 @@
+describe_tags(Opts, #aws_config{} = Config) ->
+    {AwsOpts, WorkspacesOpts} = opts(describe_tags_opts(), Opts),
+    Return = workspaces_request(
+                Config,
+                "DescribeTags",
+                AwsOpts),
+    out(Return, fun(Json, UOpts) ->
+            TagsList = proplists:get_value(<<"TagList">>, Json),
+            decode_tags_list(TagsList, UOpts)
+        end, 
+        WorkspacesOpts).
+
+%%%------------------------------------------------------------------------------
+%%% Internal Functions
+%%%------------------------------------------------------------------------------
+workspaces_request(Config, Operation, Body) ->
+    case erlcloud_aws:update_config(Config) of
+        {ok, Config1} ->
+            workspaces_request_no_update(Config1, Operation, Body);
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+workspaces_request_no_update(Config, Operation, Body) ->
+    Payload = case Body of
+               [] -> <<"{}">>;
+               _ -> jsx:encode(lists:flatten(Body))
+           end,
+    Headers = headers(Config, Operation, Payload),
+    Request = #aws_request{service = workspaces,
+                           uri = uri(Config),
+                           method = post,
+                           request_headers = Headers,
+                           request_body = Payload},
+    case erlcloud_aws:request_to_return(erlcloud_retry:request(Config, Request, fun workspaces_result_fun/1)) of
+        {ok, {_RespHeaders, <<>>}} -> {ok, []};
+        {ok, {_RespHeaders, RespBody}} -> {ok, jsx:decode(RespBody, [{return_maps, false}])};
+        {error, _} = Error-> Error
+    end.
+
+-spec workspaces_result_fun(Request :: aws_request()) -> aws_request().
+workspaces_result_fun(#aws_request{response_type = ok} = Request) ->
+    Request;
+workspaces_result_fun(#aws_request{response_type = error,
+                                  error_type = aws,
+                                  response_status = Status} = Request) when Status >= 500 ->
+    Request#aws_request{should_retry = true};
+workspaces_result_fun(#aws_request{response_type = error, error_type = aws} = Request) ->
+    Request#aws_request{should_retry = false}.
+
+headers(Config, Operation, Body) ->
+    Headers = [{"host", Config#aws_config.workspaces_host},
+               {"x-amz-target", lists:append(["WorkspacesService.", Operation])},
+               {"content-type", "application/x-amz-json-1.1"}],
+    Region = erlcloud_aws:aws_region_from_host(Config#aws_config.workspaces_host),
+    erlcloud_aws:sign_v4_headers(Config, Headers, Body, Region, "workspaces").
+
+uri(#aws_config{workspaces_scheme = Scheme, workspaces_host = Host} = Config) ->
+    lists:flatten([Scheme, Host, port_spec(Config)]).
+
+port_spec(#aws_config{workspaces_port=80}) ->
+    "";
+port_spec(#aws_config{workspaces_port=Port}) ->
+    [":", erlang:integer_to_list(Port)].
+
+
+to_binary(undefined) -> undefined;
+to_binary(true) -> true;
```

Comment:
this... isn't binary? seems kind of misleading.

Reply:
Agree. This is done for json body encoding for AWS request. I'll rename it